src/NServiceBus.Core/Pipeline/Incoming/TransportReceiveToPhysicalMessageProcessingConnector.cs

```diff
@@ -24,14 +24,14 @@ public TransportReceiveToPhysicalMessageProcessingConnector(IOutboxStorage outbo
         public override async Task Invoke(ITransportReceiveContext context, Func<IIncomingPhysicalMessageContext, Task> stage, Func<IBatchDispatchContext, Task> fork)
         {
             var messageId = context.Message.MessageId;
-            var physicalMessageContext = new IncomingPhysicalMessageContext(context.Message, context);
+            var physicalMessageContext = this.CreateIncomingPhysicalMessageContext(context);
```

Comment:
instead of hiding `context.Message` should we pass in the message and the source context?